### PR TITLE
infrastructure and repositories

### PR DESCRIPTION
**Pull request by p-Korsholm**

### CODE REVIEW

File: API/Infrastructure/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
```
Comment (BBPL) [inline API/Infrastructure/Class1.cs:1]: useless file please remove

File: API/Infrastructure/Interfaces/ITimelineRepository.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿using System;
+namespace Infrastructure.Interfaces
+{
+    public interface ITimelineRepository
```
Comment (BBPL) [inline API/Infrastructure/Interfaces/ITimelineRepository.cs:4]: I think it is a bad idea to have empty interfaces

File: API/Infrastructure/Interfaces/IUserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿using System;
+namespace Infrastructure.Interfaces
+{
+    public interface IUserRepository
```
Comment (BBPL) [inline API/Infrastructure/Interfaces/IUserRepository.cs:4]: same here

File: API/Minitwit.API/Controllers/AuthController.cs [hunk 4]
```diff
@@ -4,27 +4,64 @@
...
+    [Route("")]
     [ApiController]
     public class AuthController : ControllerBase
     {
-        // GET: api/Auth
-        [HttpGet]
-        public IEnumerable<string> Get()
+
+        [HttpGet("/login")]
+        public User LoginGet()
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/AuthController.cs]: Shouldn't this part return something more than a user?
Maybe error code 401 unauthorized? In that case it should return an IActionResult

File: API/Minitwit.API/Controllers/AuthController.cs [hunk 5]
```diff
@@ -4,27 +4,64 @@
...
+            //if logged in dont do anything
+            return null;
         }
 
-        // GET: api/Auth/5
-        [HttpGet("{id}", Name = "Get")]
-        public string Get(int id)
+
+        [HttpPost("/login")]
+        public User LoginPost()
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/AuthController.cs]: Same as above

File: API/Minitwit.API/Controllers/AuthController.cs [hunk 6]
```diff
@@ -4,27 +4,64 @@
...
         {
-            return "value";
+            //if logged in dont do anything
+            return null;
         }
 
+//@app.route('/login', methods=['GET', 'POST'])
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/AuthController.cs]: Removed Unused comments

File: API/Minitwit.API/Controllers/AuthController.cs [hunk 7]
```diff
@@ -4,27 +4,64 @@
...
+        //// GET: api/Auth/5
+        //[HttpGet("{id}", Name = "Get")]
+        //public string Get(int id)
+        //{
+        //    return "value";
+        //}
+
         // POST: api/Auth
         [HttpPost]
         public void Post([FromBody] string value)
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/AuthController.cs]: Returns IActionResult

File: API/Minitwit.API/Controllers/TimelineController.cs [hunk 8]
```diff
@@ -11,13 +12,20 @@ namespace Minitwit.API.Controllers
...
-        public IEnumerable<string> Get()
+        private readonly ITimelineRepository timelineRepository;
+
+        public TimelineController(ITimelineRepository timelineRepository)
         {
-            return new string[] { "value1", "value2" };
+            this.timelineRepository = timelineRepository;
         }
 
+        //// GET: api/Timeline
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/TimelineController.cs]: Shouldn't this be removed?

**Review by banksdev — APPROVED** [review]
this is fine

File: API/Minitwit.API/Controllers/UserController.cs [hunk 9]
```diff
@@ -11,20 +12,27 @@ namespace Minitwit.API.Controllers
...
-        // GET: api/User/5
-        [HttpGet("{id}", Name = "Get")]
-        public string Get(int id)
+        public UserController(IUserRepository userRepository)
         {
-            return "value";
+            this.userRepository = userRepository;
         }
 
+        //// GET: api/User
```
Comment (BBPL) [inline API/Minitwit.API/Controllers/UserController.cs]: Remove unused code?

File: API/Minitwit.API/Properties/launchSettings.json [hunk 10]
```diff
@@ -17,21 +17,20 @@
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
+    "Docker": {
+      "commandName": "Docker",
+      "launchBrowser": true,
+      "launchUrl": "{Scheme}://{ServiceHost}:{ServicePort}/weatherforecast",
```
Comment (BBPL) [inline API/Minitwit.API/Properties/launchSettings.json:23]: weather forecast? shouldn't it be removed?

File: API/Repository/TimelineRepository.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using Infrastructure.Interfaces;
+
+namespace Repository
+{
+    public class TimelineRepository : ITimelineRepository
+    {
```
Comment (BBPL) [inline API/Repository/TimelineRepository.cs:6]: No empty interfaces

File: API/Repository/UserRepository.cs [hunk 12]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+using Infrastructure.Interfaces;
+
+namespace Repository
+{
+    public class UserRepository : IUserRepository
+    {
```
Comment (BBPL) [inline API/Repository/UserRepository.cs:6]: same here

**Review by BBPL — CHANGES_REQUESTED** [review]

**Review by BBPL — APPROVED** [review]

**Pull request merged**